### PR TITLE
Converting IDLFragmentExtractor whatwg test to manual test

### PR DESCRIPTION
**Pull request by m-cheung**
This change is made due to memory concerns (and increasing testing time). As we add new tests, we get approach the memory limit again. In order to avoid future flaky tests (due to Node crashing from out of memory), I think it is best if parsing the whatwg HTML spec gets converted to a manual test.

### CODE REVIEW

File: test/node/parsing/IDLFragmentExtractor-manual.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('IDLFragmentExtractor-manual', function() {
+  var HTMLFileContents;
+  var IDLFragmentExtractor;
+  var HTTPRequest;
+
+  function cmpTest(testName, testDirectory, numExpectedIDLFragments, raw) {
```
Comment (arobins) [inline test/node/parsing/IDLFragmentExtractor-manual.js]: This should be shared, not copied.

File: test/node/parsing/IDLFragmentExtractor-manual.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 The Chromium Authors. ALl rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('IDLFragmentExtractor-manual', function() {
+  var HTMLFileContents;
+  var IDLFragmentExtractor;
+  var HTTPRequest;
```
Comment (arobins) [inline test/node/parsing/IDLFragmentExtractor-manual.js]: This looks unused.

**Review by arobins — COMMENTED** [review]
I'm a little concerned that running out of memory is a problem, but that's likely not under our control.

**Comment by m-cheung** [comment]
@arobins I looked at it for a couple of days a few weeks ago. There may be a few factors at play here.
1. Jasmine does not seem be freeing memory after each test is complete. This may be done as it retains data that may be returned as output at the end of all tests.
2.  Some FOAM thing that I may have overlooked may have cyclic dependencies and as a result, Javascript's GC isn't being triggered when it should.

I've spoken to @mdittmer and we believe it is best we press on for now. If you have any theories on what other factor might be contributing to this memory issue, I would love to hear them.

**Review by arobins — APPROVED** [review]
lgtm

**Pull request merged**